libass/ass_render.h

```diff
@@ -342,6 +349,21 @@ struct ass_renderer {
     BitmapEngine engine;
 
     ASS_Style user_override_style;
+
+#if ENABLE_THREADS
+    pthread_mutex_t mutex;
+    int mutex_inited;
+    pthread_cond_t main_cond, pool_cond;
+    int main_cond_inited, pool_cond_inited;
+
+    unsigned n_threads, started_threads, active_threads;
+    pthread_t *threads;
+
+    int thread_start_failed;
```

Comment:
Failure modes are total mess here: 7 status variables instead of 1&ndash;2. I feel it can be greatly simplified.